resources/tentative/complex-static-html/src/big-dom-generator/gen-css.js

```diff
@@ -1,152 +1,114 @@
-                return ".header";
-            return ".main";
-        case 3:
-            return ".todoapp";
-        case 2:
-            return ".todoholder";
-        case 1:
-            switch (index) {
-                case 0:
-                    return ".show-more";
-                case 1:
-                    return ".ribbon";
-                case 2:
-                    return ".top-bar";
-                case 3:
-                    return ".tree-area";
-                case 4:
-                    return ".todo-area";
-                default:
-                    throw new Error(`Invalid index: ${index}`);
-            }
-        case 0:
-            return ".main-ui";
-        default:
-            throw new Error(`Invalid depth: ${depth}`);
+const Combinator = {
+    DESCENDANT: " ",
+    CHILD: " > ",
+    ADJACENT_SIBLING: " + ",
+    GENERAL_SIBLING: " ~ ",
+};
+
+const getHtmlMarkup = (angular) => {
+    return angular ? ANGULAR_TODO_MVC_HTML_MARKUP : TODO_MVC_HTML_MARKUP;
+};
+
+/**
+ * List item structure for Angular:
+ * <app-todo-item>
+ *     <li class="targeted li-101">
+ *         <div class="targeted view-101"/>
+ *     </li>
+ * </app-todo-item>
+ */
+const addTodoItems = (document, NUM_TODOS_TO_INSERT_IN_HTML, angular) => {
+    const todoList = document.querySelector(".todo-list");
+
+    for (let i = 0; i < NUM_TODOS_TO_INSERT_IN_HTML; i++) {
+        const li = document.createElement("li");
+        li.className = `li-${i}`;
+
+        const div = document.createElement("div");
+        div.className = `view-${i}`;
+
+        li.appendChild(div);
+
+        if (angular) {
+            const appTodoItem = document.createElement("app-todo-item");
+
+            appTodoItem.appendChild(li);
+            todoList.appendChild(appTodoItem);
+        } else {
+            todoList.appendChild(li);
+        }
     }
 };
 
-const getType = (depth, index) => {
-    switch (depth) {
-        case 7:
-            return "div";
-        case 6:
-            return "li";
-        case 5:
-            // prettier-ignore
-            if (!index)
-                return "div";
-            return "ul";
-        case 4:
-            // prettier-ignore
-            if (!index)
-                return "header";
-            return "main";
-        case 3:
-            return "section";
-        case 2:
-        case 1:
-            return "div";
-        case 0:
-            return ".main-ui";
+const getClassname = (element) => {
+    return element && element.classList.length > 0 ? `.${element.classList[0]}` : element.nodeName.toLowerCase();
+};
+
+const getElementType = (element) => {
+    return element.nodeName.toLowerCase();
+};
+
+const getElementAtDepth = (combinator, element, currentDepth, depth) => {
+    let currentElement = element;
+    while (currentDepth > depth) {
+        currentElement = currentElement.parentElement;
+        currentDepth--;
+    }
+    return getRandomElement(combinator, currentElement);
+};
```

Comment:
No need to use the while logic for all the cases, you can write it for only the general descendant case.